zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1054,15 +1082,46 @@ public Stat getStat(final String path) {
   }
 
   private Stat getStat(final String path, final boolean watch) {
+    return getStat(path, watch, false);
+  }
+
+  /*
+   * Install watch if there is such node and return the stat
+   *
+   * If useGetData false, use exist(). @param watch would determine if adding watch
+   * to ZooKeeper server or not.
+   * Note, if @param path does not exist in ZooKeeper server, watch would still be installed
+   * if @param watch is true.
+   *
+   * If useGetData true, use getData() to add watch. ignore @param watch in this case.
+   * Note, if @param path does not exist in ZooKeeper server, no watch would be added.
+   */
+  private Stat getStat(final String path, final boolean watch, final boolean useGetData) {
```

Comment:
I thought about this method, I think instead of having this "useGetData" option, we just make it explicitly getData(final String path, final boolean watch) method. The caller know which one to all. And then there won't be any confusion.

Reply:
add a todo. @jiajunwang can help to refactor later.

Reply:
Why?